sdk/storage/azure-storage-blob-nio/src/main/java/com/azure/storage/blob/nio/AzureFileSystemProvider.java

```diff
@@ -785,11 +788,14 @@ public FileStore getFileStore(Path path) throws IOException {
     }
 
     /**
-     * Unsupported.
+     * Checks the existence, and optionally the accessibility, of a file.
+     * <p>
+     * This method may only be used to check the existence of a file. It is not possible to determine the permissions
+     * granted to a given client, so if any mode argument is specified, an {@link UnsupportedOperationException} will be
```

Comment:
Should this be UnsupportedOperationException or AccessDeniedException?

Reply:
AccessDenied

Reply:
(per the jdk docs for this method: "the requested access would be denied or the access cannot be determined because the Java virtual machine has insufficient privileges or other reasons." I think it's reasonable to put this under access cannot be determined)